### PR TITLE
add limit executor

### PR DESCRIPTION
**Pull request by lishihai9017**
hi,

this pr add limit executor.

### CODE REVIEW

File: src/server/coprocessor/executor/limit.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    use util::codec::table;
+    use util::codec::number::NumberEncoder;
+
+    use tipb::schema::ColumnInfo;
+    use tipb::executor::TableScan;
+
+    use protobuf::RepeatedField;
+
+    // the first column should be i64 since it will be used as row handle
+    pub fn gen_table_data(tid: i64,
```
Comment (andelf) [inline src/server/coprocessor/executor/limit.rs]: ~pub~

File: src/server/coprocessor/executor/limit.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    use util::codec::number::NumberEncoder;
+
+    use tipb::schema::ColumnInfo;
+    use tipb::executor::TableScan;
+
+    use protobuf::RepeatedField;
+
+    // the first column should be i64 since it will be used as row handle
+    fn gen_table_data(tid: i64,
+                          cis: &[ColumnInfo],
```
Comment (andelf) [inline src/server/coprocessor/executor/limit.rs]: indentation(format check will fail)

File: src/server/coprocessor/executor/limit.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+struct LimitExecutor<'a> {
+    limit: u64,
+    cursor: u64,
+    src: Box<Executor + 'a>,
+}
+
+impl<'a> LimitExecutor<'a> {
+    fn new(limit: Limit, src: Box<Executor + 'a>) -> LimitExecutor {
+        LimitExecutor {
+            limit: limit.get_limit() as u64,
```
Comment (andelf) [inline src/server/coprocessor/executor/limit.rs]: ~as u64~

**Comment by AndreMouche** [comment]
Please resolve conflicts.

**Comment by siddontang** [comment]
PTAL @hhkbp2 @andelf

**Pull request merged**